runtime/compiler/optimizer/MethodHandleTransformer.cpp

```diff
@@ -773,3 +776,155 @@ TR_MethodHandleTransformer::process_java_lang_invoke_Invokers_checkCustomized(TR
+                                                                                 TR::Address,
+                                                                                 mhTableOffset,
+                                                                                 false,
+                                                                                 true,
+                                                                                 true,
+                                                                                 mhTableFieldNameAndSig);
+         TR::Node *mhTableNode = TR::Node::createWithSymRef(comp()->il.opCodeForIndirectLoad(TR::Address), 1, 1, mhTableHolder, mhTableSymRef);
+         mhTableNode->copyByteCodeInfo(node);
+
+         // get index for load
+         uintptr_t accessDescriptorObj = knot->getPointer(adIndex);
+         int32_t mhEntryIndex = comp()->fej9()->getInt32Field(accessDescriptorObj, "mode");
+         TR::Node *arrayIndexNode = TR::Node::create(TR::iconst, 0, mhEntryIndex);
+         arrayIndexNode->copyByteCodeInfo(node);
+
+         // calculate element address, create load from array, and refine MH symref with known object info
+         TR::Node *mhAddressNode = J9::TransformUtil::calculateElementAddress(comp(), mhTableNode, arrayIndexNode, TR::Address);
+         TR::Node *mhNode = TR::Node::createWithSymRef(comp()->il.opCodeForIndirectArrayLoad(TR::Address), 1, 1, mhAddressNode, comp()->getSymRefTab()->findOrCreateArrayShadowSymbolRef(TR::Address, mhTableNode));
+         mhNode->copyByteCodeInfo(node);
+         TR::SymbolReference *refinedMHSymRef = comp()->getSymRefTab()->findOrCreateSymRefWithKnownObject(mhNode->getSymbolReference(), mhIndex);
+         mhNode->setSymbolReference(refinedMHSymRef);
+
+         // Insert spine check for arraylets
+         if (TR::Compiler->om.canGenerateArraylets())
+            {
+            TR::SymbolReference * spineCHKSymRef = comp()->getSymRefTab()->findOrCreateArrayBoundsCheckSymbolRef(comp()->getMethodSymbol());
+            TR::Node * spineCHKNode = TR::Node::create(node, TR::SpineCHK, 3);
+            spineCHKNode->setAndIncChild(0, mhNode);
+            spineCHKNode->setAndIncChild(1, mhTableNode);
+            spineCHKNode->setAndIncChild(2, arrayIndexNode);
+            spineCHKNode->setSymbolReference(spineCHKSymRef);
+            spineCHKNode->setSpineCheckWithArrayElementChild(true, comp());
+            tt->insertBefore(TR::TreeTop::create(comp(), spineCHKNode));
+            }
+
+         if (comp()->useCompressedPointers())
+            tt->insertBefore(TR::TreeTop::create(comp(), TR::Node::createCompressedRefsAnchor(mhNode)));
+         else
+            tt->insertBefore(TR::TreeTop::create(comp(), TR::Node::create(node, TR::treetop, 1, mhNode)));
+
+         TR::TransformUtil::transformCallNodeToPassThrough(this, node, tt, mhNode);
```

Comment:
I think `transformCallNodeToPassThrough()` is intended for cases where the result of the call is one of its children (or at least one of its descendants). In this case, though, the result is coming from a freshly created load that's meant to be evaluated here at the point of evaluation of the call. IMO it would be better to transmute `node` into `mhNode` directly so that we don't need to use `PassThrough`